### DISCUSSION TITLE
While Linking OpenROAD, undefined reference to TINFO

### DISCUSSION THREAD

**Discussion by cubbins-git** [The-OpenROAD-Project/OpenROAD-flow-scripts#1879] [Q&A] [answered]
I am trying to build OpenROAD. All appears working except the final step where the build faults, with an undefined reference to a link library, TINFO. The standard solution says to add '-lreadline -ltinfo' to a file, yet I am not able to determine which file I should add. Which file to edit? Adding '-ltinfo' The error message is below.

[ 95%] Built target par
[ 97%] Linking CXX executable openroad
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `PC@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `tputs@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `tgoto@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `tgetflag@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `tgetnum@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `UP@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `tgetent@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `tgetstr@NCURSES6_TINFO_5.0.19991023'
/usr/bin/ld: /lib/x86_64-linux-gnu/libreadline.so.8: undefined reference to `BC@NCURSES6_TINFO_5.0.19991023'

**Comment by maliberty** [comment]
What platform and compiler are you using?  This seems like a local system issue.

**Reply by cubbins-git** [reply, marked as answer]
Thank you for your early response. I have it now. All is well. I likely had some local issues as you explained. I can see now, having a clean install of Ubuntu and new OpenROAD git clone, is the best way to approach a large compilation such as found in OpenROAD.
